docs/site/tutorials/todo-list/todo-list-tutorial-has-one-relation.md

```diff
@@ -0,0 +1,255 @@
+---
+
+We have that a `Todo` [`belongsTo`](../../BelongsTo-relation.md) a `TodoList`
+and a `TodoList` [`hasMany`](../../HasMany-relation.md) `Todo`s. Another type of
+relation we can add is [`hasOne`](../../hasOne-relation.md). To do so, let's add
+`TodoListImage` such that each `TodoList` `hasOne` image.
+
+### Create the Model
+
+Similar to how we created the model for
+[`TodoList`](todo-list-tutorial-model.md), using `lb4 model`:
+
+```sh
+lb4 model
+? Model class name: TodoListImage
+? Please select the model base class Entity (A persisted model with an ID)
+? Allow additional (free-form) properties? No
+Model TodoListImage will be created in src/models/todo-list-image.model.ts
+
+Let's add a property to TodoListImage
+Enter an empty property name when done
+
+? Enter the property name: id
+? Property type: number
+? Is id the ID property? Yes
+? Is id generated automatically? No
+? Is it required?: No
+? Default value [leave blank for none]:
+
+Let's add another property to TodoListImage
+Enter an empty property name when done
+
+? Enter the property name: value
+? Property type: string
+? Is it required?: Yes
+? Default value [leave blank for none]:
+
+Let's add another property to TodoListImage
+Enter an empty property name when done
+
+? Enter the property name:
+   create src/models/todo-list-image.model.ts
+   update src/models/index.ts
+
+Model TodoListImage was created in src/models/
+```
+
+### Create the Repository
+
+Using `lb4 repository`, let's create the repository:
+
+```sh
+lb4 repository
+? Please select the datasource DbDatasource
+? Select the model(s) you want to generate a repository TodoListImage
+? Please select the repository base class DefaultCrudRepository (Legacy juggler bridge)
+   create src/repositories/todo-list-image.repository.ts
+   update src/repositories/index.ts
+
+Repository TodoListImageRepository was created in src/repositories/
+```
+
+### Add the Relation
+
+First, let's add the relation to the model classes:
+
+{% include code-caption.html content="src/models/todo-list-image.model.ts" %}
+
+```ts
+import {belongsTo} from '@loopback/repository';
+import {TodoList, TodoListWithRelations} from './todo-list.model';
+
+@model()
+export class TodoListImage extends Entity {
+  // ... other properties
+
+  @belongsTo(() => TodoList)
+  todoListId: number;
+
+  // ...
+}
+
+export interface TodoListImageRelations {
+  todoList?: TodoListWithRelations;
+}
+```
+
+{% include code-caption.html content="src/models/todo-list.model.ts" %}
+
+```ts
+import {hasOne} from '@loopback/repository';
+import {
+  TodoListImage,
+  TodoListImageWithRelations,
+} from './todo-list-image.model';
+
+@model()
+export class TodoList extends Entity {
+  // ... other properties
+
+  @hasOne(() => TodoListImage)
+  image?: TodoListImage;
+
+  // ...
+}
+
+export interface TodoListRelations {
+  todos?: TodoWithRelations[];
+
+  // Add the following line
+  image?: TodoListImageWithRelations;
+}
+```
+
```

Comment:
It's nice to have this page! Maybe we can explain here that the CLI `lb4 relation` doesn't support `hasOe` yet. But it's easy to set up.

Also, could you add links to Tutorial [Model](https://loopback.io/doc/en/lb4/Model.html), [Repo](https://loopback.io/doc/en/lb4/Repositories.html), [Datasource](https://loopback.io/doc/en/lb4/DataSources.html), and [Controller](https://loopback.io/doc/en/lb4/Controllers.html) page respectively and briefly explain that check out these page to learn how to configure all these artifacts such as customizing names and DB configurations ( something similar to the **Note** in [Relation Tutorial](https://loopback.io/doc/en/lb4/todo-list-tutorial-relations.html) page).

Reply:
> It's nice to have this page! Maybe we can explain here that the CLI lb4 relation doesn't support hasOe yet. But it's easy to set up.

I mentioned it at the end of the section, but I can move it to the top 👍